apps/examples/testcase/ta_tc/eventloop/utc/utc_eventloop_main.c

```diff
@@ -0,0 +1,211 @@
+{
+	printf("Delete timer callback is called!\n");
+}
+
+static void utc_eventloop_loop_run_n(void)
+{
+}
+
+static void utc_eventloop_loop_run_p(void)
+{
+	int ret;
+
+	ret = eventloop_loop_run();
+	TC_ASSERT_EQ("eventloop_loop_run", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_loop_stop_n(void)
+{
+}
+
+static void utc_eventloop_loop_stop_p(void)
+{
+	int ret;
+
+	ret = eventloop_loop_stop();
+	TC_ASSERT_EQ("eventloop_loop_stop", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_add_timer_n(void)
+{
+	timer = eventloop_add_timer(1000, 0, NULL, NULL);
+	(void)eventloop_loop_run();
+	TC_ASSERT_EQ("eventloop_add_timer", timer, NULL);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_add_timer_p(void)
+{
+	char *timer_data;
+	char *timer_repeat_data;
+
+	timer_data = malloc(sizeof(strlen(EL_TIMER_DATA)));
+	timer_repeat_data = malloc(sizeof(strlen(EL_TIMER_REPEAT_DATA)));
+	strncpy(timer_data, EL_TIMER_DATA, strlen(EL_TIMER_DATA) + 1);
+	strncpy(timer_repeat_data, EL_TIMER_REPEAT_DATA, strlen(EL_TIMER_REPEAT_DATA) + 1);
+	
+	el_timer_flag = 0;
+	timer = eventloop_add_timer(0, 0, el_timer_cb, timer_data);
+	(void)eventloop_loop_run();
+	TC_ASSERT_EQ_CLEANUP("eventloop_add_timer", el_timer_flag, 1, free(timer_data));
+
+	el_timer_flag = 0;
+	timer_repeat = eventloop_add_timer(1000, 1, el_timer_repeat_cb, timer_repeat_data);
+	(void)eventloop_loop_run();
+	TC_ASSERT_EQ_CLEANUP("eventloop_add_timer", el_timer_flag, 3, free(timer_repeat_data));
+
+	free(timer_data);
+	free(timer_repeat_data);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_delete_timer_n(void)
+{
+	int ret;
+
+	ret = eventloop_delete_timer(NULL);
+	TC_ASSERT_EQ("eventloop_delete_timer", ret, EVENTLOOP_INVALID_PARAM);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_delete_timer_p(void)
+{
+	el_timer_t *del_timer;
+	int ret;
+	
+	el_timer_flag = 0;
+	del_timer = eventloop_add_timer_async(0, 0, el_delete_timer_cb, NULL);
+	ret = eventloop_delete_timer(del_timer);
+	
+	TC_ASSERT_EQ("eventloop_delete_timer", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_add_timer_async_n(void)
+{
+	timer = eventloop_add_timer_async(1000, 0, NULL, NULL);
+	TC_ASSERT_EQ("eventloop_add_timer_async", timer, NULL);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_add_timer_async_p(void)
+{
+	int sleep_cnt;
+	char *timer_data;
+	char *timer_repeat_data;
+
+	timer_data = malloc(sizeof(strlen(EL_TIMER_DATA)));
+	timer_repeat_data = malloc(sizeof(strlen(EL_TIMER_REPEAT_DATA)));
+	strncpy(timer_data, EL_TIMER_DATA, strlen(EL_TIMER_DATA) + 1);
+	strncpy(timer_repeat_data, EL_TIMER_REPEAT_DATA, strlen(EL_TIMER_REPEAT_DATA) + 1);
+
+	el_timer_flag = 0;
+	timer = eventloop_add_timer_async(0, 0, el_timer_cb, timer_data);
+	sleep_cnt = 0;
+	while (sleep_cnt < 5) {
+		sleep(1);
+		sleep_cnt++;
+		if (el_timer_flag == 1) {
+			break;
+		}
+	}
+	TC_ASSERT_EQ("eventloop_add_timer_async", el_timer_flag, 1);
```

Comment:
Please check free for allocated variable

Reply:
Done